junit-platform-console/src/main/java/org/junit/platform/console/options/Theme.java

```diff
@@ -79,43 +79,43 @@ public static Theme valueOf(Charset charset) {
 		this.blank = new String(new char[vertical().length()]).replace('\0', ' ');
 	}
 
-	public String root() {
+	public final String root() {
```

Comment:
> `Theme` is final anyway, isn't it?

Yeah, I wondered about that...

> Why these?

The reason is a bit complicated, so please bear with me. :)

Before we can [successfully adopt error-prone](https://github.com/junit-team/junit5/pull/961), we need to make sure that all error-prone checks that we have enabled pass, because error-prone is an OpenJDK-derived compiler developed by Google that makes additional compile-time guarantees over the other javac compilers out there, and so if we want things to compile successfully, error-prone needs to be appeased.

We don't strictly need the `final`s for most of the methods in `Theme`, but for whatever reason, we need it at least on `Theme.vertical()`, because error-prone sees it as an overridable method that is being called by a constructor - `Theme`'s constructor specifically. In most cases, i.e. with non-final classes, this would be a real security bug, because then malicious users could subclass and change the behaviour of `vertical()`, thus changing the behaviour of the constructor and causing to be in an invalid state.

However, as you realised, since `Theme` is an enum, it's effectively final.

It seems that error-prone isn't intelligent enough yet to realise that, when an _enum_ constructor calls an overridable method, it can be safely ignored since enums cannot be subclassed. I suspect that error-prone takes a liberal approach where it considers _every_ class or class-like structure to be non-final, and for this reason it mandates at compile-time that `Theme.vertical()` _must_ be `final`.

I should really submit it as a bug report to the error-prone guys, but in the short-term at least, I thought it sensible to make not _only_ `Theme.vertical()` `final`, but every other method as well, to make things look consistent.

It's not ideal, yes, but I currently cannot see a way forward other than disabling the error-prone rule that mandates the `final` keyword's existence (where the rule is http://errorprone.info/bugpattern/ConstructorInvokesOverridable), which I consider to be the worse option of the two.

I hope this clarifies things for you @marcphilipp.